cartoframes/dataset.py

```diff
@@ -39,34 +39,38 @@ class Dataset(object):
 
     DEFAULT_RETRY_TIMES = 3
 
-    def __init__(self, table_name=None, schema='public', query=None, df=None, gdf=None, state=None, context=None):
-        self.table_name = normalize_name(table_name)
-        self.schema = schema
-        self.query = query
-        self.df = df
-        self.gdf = gdf
+    def __init__(self, table_name=None, schema='public',
+                 query=None, df=None, gdf=None,
+                 state=None, is_saved_in_carto=False, context=None):
+        self._table_name = normalize_name(table_name)
+        self._schema = schema
+        self._query = query
+        self._df = df
+        self._gdf = gdf
 
         if not self._validate_init():
             raise ValueError('Wrong Dataset creation. You should use one of the class methods: '
                              'from_table, from_query, from_dataframe, from_geodataframe, from_geojson')
 
-        self.state = state
-        self.cc = context or default_context
+        self._cc = context or default_context
+        self._state = state
+        self._is_saved_in_carto = is_saved_in_carto
 
-        self.normalized_column_names = None
-        if self.df is not None:
-            _save_index_as_column(self.df)
-            self.normalized_column_names = _normalize_column_names(self.df)
-        elif self.gdf is not None:
-            _save_index_as_column(self.gdf)
-            self.normalized_column_names = _normalize_column_names(self.gdf)
+        self._normalized_column_names = None
+        if self._df is not None:
+            _save_index_as_column(self._df)
+            self._normalized_column_names = _normalize_column_names(self._df)
+        elif self._gdf is not None:
+            _save_index_as_column(self._gdf)
+            self._normalized_column_names = _normalize_column_names(self._gdf)
```

Comment:
I am just changing this for other reason https://github.com/CartoDB/cartoframes/pull/688

About having just one variable `self._df`,  I agree. But we will need to find the best way to do it because we will have to change some important methods (upload, download, ...)

Reply:
I see, then I think we can just include another bullet to https://github.com/CartoDB/cartoframes/issues/670

Reply:
I am going to create a issue about it with the `1.0` label

Reply:
Done

Reply:
Going green?

Reply:
![](https://preview.redd.it/g7tj62rann7x.jpg?width=960&crop=smart&auto=webp&s=2bf990da3d96034a6c3a24bd3c5448416aa4e666)